pkg/ccloudv2/srcm.go

```diff
@@ -103,10 +118,34 @@ func (c *Client) GetSchemaRegistryClustersByEnvironment(environment string) ([]s
 	return list, nil
 }
 
+func (c *Client) GetSrcmV3ClustersByEnvironment(environment string) ([]srcmv3.SrcmV3Cluster, error) {
+	var list []srcmv3.SrcmV3Cluster
+
+	done := false
+	pageToken := ""
+	for !done {
+		req := c.SrcmV3Client.ClustersSrcmV3Api.ListSrcmV3Clusters(c.srcmV3ApiContext()).Environment(environment)
+		if pageToken != "" {
+			req = req.PageToken(pageToken)
+		}
+		page, httpResp, err := c.SrcmV3Client.ClustersSrcmV3Api.ListSrcmV3ClustersExecute(req)
+		if err != nil {
+			return nil, errors.CatchCCloudV2Error(err, httpResp)
+		}
+		list = append(list, page.GetData()...)
+
+		pageToken, done, err = extractNextPageToken(page.GetMetadata().Next)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return list, nil
+}
+
 func (c *Client) executeListSchemaRegistryClusters(environment, pageToken string) (srcmv2.SrcmV2ClusterList, *http.Response, error) {
-	req := c.SrcmClient.ClustersSrcmV2Api.ListSrcmV2Clusters(c.srcmApiContext()).Environment(environment)
+	req := c.SrcmV2Client.ClustersSrcmV2Api.ListSrcmV2Clusters(c.srcmApiContext()).Environment(environment)
 	if pageToken != "" {
 		req = req.PageToken(pageToken)
 	}
-	return c.SrcmClient.ClustersSrcmV2Api.ListSrcmV2ClustersExecute(req)
+	return c.SrcmV2Client.ClustersSrcmV2Api.ListSrcmV2ClustersExecute(req)
 }
```

Comment:
An alternative implementation is to just replace this srcm/v2 API call with srcm/v3 instead of creating a separate function to call srcm/v3, but that would affect a bunch of other components. Is it worth the risk?

Reply:
> but that would affect a bunch of other components

What do you mean by this?